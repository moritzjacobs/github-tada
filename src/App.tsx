import React from "react";
import Button from "./Button";
import {Comment} from "./comments";

type Props = {
	comments: Array<Comment>,
};

const App = ({comments}: Props) => {
	return (
		<>
			<div className="discussion-sidebar-heading text-bold">
				Browse by reaction
			</div>
			{comments.map(c => (
				<Button key={c.target.dataset.gid} target={c.target} reactions={c.reactions} />
			))}
		</>
	);
};

export default App;
