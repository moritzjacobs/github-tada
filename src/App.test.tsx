import React from "react";
import renderer from "react-test-renderer";
import App from "./App";

const reactions = [
	{
		emoji: "ðŸ’¯",
		type: "100",
		count: 23,
	},
];

const target = document.createElement("div");

target.dataset.gid = "1234";

describe("<App/>", () => {
	it("renders correctly when there are no items", () => {
		const tree = renderer.create(<App comments={[]} />).toJSON();

		expect(tree).toMatchSnapshot();
	});
	it("renders correctly when there are items", () => {
		const tree = renderer.create(<App comments={[{target, reactions}]} />).toJSON();

		expect(tree).toMatchSnapshot();
	});
});
