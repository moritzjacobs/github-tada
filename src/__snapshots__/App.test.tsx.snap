// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> renders correctly when there are no items 1`] = `null`;

exports[`<App/> renders correctly when there are no items 2`] = `
Array [
  <div
    className="discussion-sidebar-heading text-bold"
  >
    Browse by reaction
  </div>,
  <button
    className="grb-button"
    onClick={[Function]}
    type="button"
  >
    <span
      className="grb-reaction"
    >
      ðŸ’¯
      Â 
      23
    </span>
  </button>,
]
`;
